crates/sui-indexer/src/types.rs

```diff
@@ -254,20 +256,25 @@ impl CheckpointTransactionBlockResponse {
         let created = self.effects.created().iter();
         let mutated = self.effects.mutated().iter();
         let unwrapped = self.effects.unwrapped().iter();
-        created
+        let recipients: Vec<String> = created
             .chain(mutated)
             .chain(unwrapped)
             .filter_map(|obj_ref| match obj_ref.owner {
-                Owner::AddressOwner(address) => Some(Recipient {
-                    id: None,
-                    transaction_digest: self.digest.to_string(),
-                    checkpoint_sequence_number: checkpoint as i64,
-                    epoch: epoch as i64,
-                    sender: self.transaction.data.sender().to_string(),
-                    recipient: address.to_string(),
-                }),
+                Owner::AddressOwner(address) => Some(address.to_string()),
                 _ => None,
             })
+            .collect();
+        let unique_recipients: Vec<String> = recipients.into_iter().unique().collect();
+        unique_recipients
+            .into_iter()
```

Comment:
You can avoid the extra allocation here
```suggestion
         recipients.into_iter().unique().map(...
```